jerry-core/lit/lit-char-helpers.c

```diff
@@ -289,32 +289,32 @@ lit_char_hex_to_int (ecma_char_t c) /**< code unit, corresponding to
  * @return true if decoding was successful, false otherwise
  */
 bool
-lit_read_code_point_from_hex (lit_utf8_byte_t *buf_p, /**< buffer with characters */
-                              lit_utf8_size_t number_of_characters, /**< number of characters to be read */
-                              lit_code_point_t *out_code_point_p) /**< [out] decoded result */
+lit_read_code_unit_from_hex (lit_utf8_byte_t *buf_p, /**< buffer with characters */
+                             lit_utf8_size_t number_of_characters, /**< number of characters to be read */
+                             ecma_char_ptr_t out_unit_point_p) /**< [out] decoded result */
 {
-  lit_code_point_t code_point = 0;
+  ecma_char_t code_unit = LIT_BYTE_NULL;
```

Comment:
This seems to be a bit of an abuse of `LIT_BYTE_NULL`. Its documentation in lit-strings.h states that it's an utf-8 string end marker. Too bad that in the same header, only `LIT_UNICODE_CODE_POINT_NULL` is defined but no `LIT_UTF16_CODE_UNIT_NULL`. What about adding it (and making use of it here)?